src/Views/Homepage.vala

```diff
@@ -292,10 +292,12 @@ public class AppCenter.Homepage : AbstractView {
 #if POP_OS
         string[] added_picks = {};
 
-        foreach (var id in newest_ids) {
-            if (package_count >= MAX_PACKAGES_IN_BANNER) {
-                break;
+        while (picks_carousel.get_children ().length () < MAX_PACKAGES_IN_BANNER) {
+            var random_pick = get_random_pick ();
+            if (random_pick in added_picks) {
```

Comment:
Rather than blindly picking from a list and adding it to a new list, and trying again if it is already in the list, the array should be sorted using a random function, then truncated.

https://valadoc.org/glib-2.0/GLib.Array.sort.html
https://stackoverflow.com/questions/64925666/how-can-i-sort-an-array-randomly-in-javascript

Reply:
As of e01fd2d I've implemented a random sort of the list instead, which should address this.